controllers/devworkspace/solver/che_routing.go

```diff
@@ -332,33 +350,115 @@ func (c *CheRoutingSolver) getGatewayConfigsAndFillRoutingObjects(cheCluster *ch
 
 	configs := make([]corev1.ConfigMap, 0)
 
+	useLegacyRouting := false
+	username, err := getNormalizedUsername(c.client, routing.Namespace)
+	if err != nil {
+		useLegacyRouting = true
+	}
+
+	dwName, err := getNormalizedWkspName(c.client, routing.Namespace, routing.Name)
+	if err != nil {
+		useLegacyRouting = true
+	}
+
+	var endpointStrategy EndpointStrategy
+	if useLegacyRouting {
+		endpointStrategy = Legacy{workspaceID: workspaceID}
```

Comment:
We don't have `componentName` here because we are not iterating over the componentsName/endpoints until the `exposeAllEndpoints` function here:

https://github.com/eclipse-che/che-operator/blob/00b627d0161d3616d0cb22dc2cbed9072f03156e/controllers/devworkspace/solver/che_routing.go#L382

https://github.com/eclipse-che/che-operator/blob/00b627d0161d3616d0cb22dc2cbed9072f03156e/controllers/devworkspace/solver/che_routing.go#L486-L496

, but I can see how this might cause some confusion,

I've added a new commit https://github.com/eclipse-che/che-operator/pull/1672/commits/bab8f187245224695c696566b0ba78d7d061058f to address it by removing the field, WDYT?